samples/bluetooth/alexa_gadget/src/gadgets/CMakeLists.txt

```diff
@@ -0,0 +1,192 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+# The Gadgets Profile relies on the following external dependencies that
+# must be downloaded and run before the build stage:
+# - nanopb generator
+# - protocol buffer definitions
+
+# Note: as there doesn't seem to be an easy way of checking the version of the nanopb generator (protoc-gen-nanopb binary)
+# there is the possibility of a version mismatch between the generator and the nanopb source files that is used in the firmware sample.
+# If there is a mismatch, there will be a compilation error in the build step.
+# The easiest way to fix such an error is to download the appropriate nanopb generator .zip and extract it to this sample folder.
+# (https://jpa.kapsi.fi/nanopb/download/)
+
+# Check local folder for protoc and nanopb binaries presence
+file(GLOB_RECURSE LOCAL_PROTOC_BIN "${PROJECT_SOURCE_DIR}/*/generator-bin/protoc")
```

Comment:
This hides any cached variable, resulting in user cannot do:
```
$ cmake -DBOARD=nrf52840dk_nrf52840 -DLOCAL_PROTOC_BIN=/projects/github/ncs/nanopb/generator/protoc -DLOCAL_NANOPB_BIN=/projects/github/ncs/nanopb/generator/protoc-gen-nanopb -GNinja -Bbuild .
Including boilerplate (Zephyr workspace): /projects/github/ncs/zephyr/cmake/app/boilerplate.cmake
-- Application: /projects/github/ncs/nrf/samples/bluetooth/alexa_gadget
...
-- nanopb not found.
-- Please download nanopb from here https://jpa.kapsi.fi/nanopb/download/
-- and extract to either this sample directory or somewhere cmake can find it via system PATH.
```

When we have the repo as part of `west`, then it should be much easier to look up the needed tools.